go.work

```diff
@@ -1,4 +1,4 @@
-go 1.22.8
```

Comment:
Do you want this go version update here?

Reply:
it's written in the PR description..
this PR depends on the go upgrade, this will be removed once I rebased after the upgrade is merged

Reply:
I updated the description to make it more clear, since I expect further reviewers to ask the same question